command/job_history.go

```diff
@@ -222,15 +262,14 @@ func parseVersion(input string) (uint64, bool, error) {
 func (c *JobHistoryCommand) formatJobVersions(versions []*api.Job, diffs []*api.JobDiff, full bool) error {
 	vLen := len(versions)
 	dLen := len(diffs)
-	if dLen != 0 && vLen != dLen+1 {
-		return fmt.Errorf("Number of job versions %d doesn't match number of diffs %d", vLen, dLen)
-	}
 
 	for i, version := range versions {
 		var diff *api.JobDiff
 		var nextVersion uint64
-		if i+1 <= dLen {
+		if dLen > i {
```

Comment:
This still feels fuzzy to me. is `dLen >= vLen` or `vLen >= dLen`? I would expect there'd potentially be more versions than diffs, but maybe I'm misunderstanding what to expect here. In which case, we'd end up dereferencing past the last diff and panic.

Maybe we need a test to exercise this function directly?

Reply:
I'll definitely have a test for this, but it previously made sure that "versions is at least one fewer than the last of the diffs", or in other words, "The last version (version 0) never gets a diff attached to it" (because before this PR, all diffs were against their preceding version, and version 0 didn't have a known preceding version)

That said: we assigned a `nextVersion` variable here and never used it — the line that was using it has been commented out for several years. I think I'll delete that and put the logic back to the way it was before this PR for this particular condition.